duckduckgo/Android
# add HttpLoggingInterceptor for debug build variants

Task/Issue URL: 
Tech Design URL: 
CC: 

**Description**:

Added the okhttp3 logging-interceptor dependency to debug build variants only. Release build variants don't need that dependency and just impl. a no-op interceptor instead.
This allows debugging api calls via logcat (which can be handy if you forgot to run debugger / connect Charles proxy etc.)

Alternatively could have added the logging interceptor to all build variants and for release builds set the log level to `NONE`, however I find it better separated from release build variants via debugImplementation.

Some example logs how it would look like:
````2020-11-01 10:08:29.393 18108-18149/com.duckduckgo.mobile.android.debug I/okhttp.OkHttpClient: --> GET https://staticcdn.duckduckgo.com/trackerblocking/v2.1/tds.json
2020-11-01 10:08:29.654 18108-18161/com.duckduckgo.mobile.android.debug I/okhttp.OkHttpClient: --> GET https://improving.duckduckgo.com/t/ml_android_phone?atb=v245-5zs&appVersion=5.69.0&test=1
2020-11-01 10:08:29.821 18108-18169/com.duckduckgo.mobile.android.debug I/okhttp.OkHttpClient: --> GET https://duckduckgo.com/atb.js?at=app_use&atb=v245-5zs&set_atb=v245-5&test=1
2020-11-01 10:08:30.509 18108-18161/com.duckduckgo.mobile.android.debug I/okhttp.OkHttpClient: <-- 200 https://improving.duckduckgo.com/t/ml_android_phone?atb=v245-5zs&appVersion=5.69.0&test=1 (854ms, 43-byte body)
2020-11-01 10:08:30.536 18108-18149/com.duckduckgo.mobile.android.debug I/okhttp.OkHttpClient: <-- 200 https://staticcdn.duckduckgo.com/trackerblocking/v2.1/tds.json (1141ms, unknown-length body)
2020-11-01 10:08:30.576 18108-18169/com.duckduckgo.mobile.android.debug I/okhttp.OkHttpClient: <-- 200 https://duckduckgo.com/atb.js?at=app_use&atb=v245-5zs&set_atb=v245-5&test=1 (754ms, unknown-length body)
2020-11-01 10:08:31.459 18108-18149/com.duckduckgo.mobile.android.debug I/okhttp.OkHttpClient: --> GET https://duckduckgo.com/contentblocking/trackers-whitelist-temporary.txt
2020-11-01 10:08:31.499 18108-18149/com.duckduckgo.mobile.android.debug I/okhttp.OkHttpClient: <-- 200 https://duckduckgo.com/contentblocking/trackers-whitelist-temporary.txt (39ms, unknown-length body)
2020-11-01 10:08:31.502 18108-18149/com.duckduckgo.mobile.android.debug I/okhttp.OkHttpClient: --> GET https://duckduckgo.com/contentblocking.js?l=surrogates
2020-11-01 10:08:31.534 18108-18149/com.duckduckgo.mobile.android.debug I/okhttp.OkHttpClient: <-- 200 https://duckduckgo.com/contentblocking.js?l=surrogates (31ms, unknown-length body)
2020-11-01 10:08:31.546 18108-18149/com.duckduckgo.mobile.android.debug I/okhttp.OkHttpClient: --> GET https://staticcdn.duckduckgo.com/https/https-mobile-v2-bloom-spec.json
2020-11-01 10:08:31.586 18108-18149/com.duckduckgo.mobile.android.debug I/okhttp.OkHttpClient: <-- 200 https://staticcdn.duckduckgo.com/https/https-mobile-v2-bloom-spec.json (39ms, unknown-length body)
2020-11-01 10:08:31.645 18108-18149/com.duckduckgo.mobile.android.debug I/okhttp.OkHttpClient: --> GET https://staticcdn.duckduckgo.com/https/https-mobile-v2-false-positives.json
2020-11-01 10:08:31.671 18108-18149/com.duckduckgo.mobile.android.debug I/okhttp.OkHttpClient: <-- 200 https://staticcdn.duckduckgo.com/https/https-mobile-v2-false-positives.json (25ms, unknown-length body)
2020-11-01 10:08:31.699 18108-18169/com.duckduckgo.mobile.android.debug I/okhttp.OkHttpClient: --> GET https://improving.duckduckgo.com/t/m_https_nl_android_phone?atb=v245-5zs&appVersion=5.69.0&test=1
2020-11-01 10:08:31.730 18108-18169/com.duckduckgo.mobile.android.debug I/okhttp.OkHttpClient: <-- 200 https://improving.duckduckgo.com/t/m_https_nl_android_phone?atb=v245-5zs&appVersion=5.69.0&test=1 (30ms, 43-byte body)
2020-11-01 10:08:31.781 18108-18149/com.duckduckgo.mobile.android.debug I/okhttp.OkHttpClient: --> GET https://staticcdn.duckduckgo.com/survey/v2/survey-mobile.json
2020-11-01 10:08:31.875 18108-18149/com.duckduckgo.mobile.android.debug I/okhttp.OkHttpClient: <-- 200 https://staticcdn.duckduckgo.com/survey/v2/survey-mobile.json (91ms, unknown-length body)
````

This PR is just a suggestion, feel free to merge or close however you prefer :wink: 

**Steps to test this PR**:
1. launch dgg debug / release variant
1. filter logcat by tag `okhttp`
1. see logs of api calls in debug variant / no logs in release variant


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
